src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Markup/Baml2006/Baml2006Reader.cs

```diff
@@ -2085,7 +2085,7 @@ private string Logic_GetFullXmlns(string uriInput)
                     else
                     {
                         int assemblyKeywordStartIdx = semicolonIdx + 1;
-                        int equalIdx = uriInput.IndexOf("=");
+                        int equalIdx = uriInput.IndexOf('=');
```

Comment:
The `assembly=` check could be also done with a more efficient string.Compare check (and only search for the '=' in case of failure to get the proper error message) - no need for any substring allocation.

Reply:
And probably don't need a substring allocation for the final empty assembly name check also.